config/confighttp/confighttp.go

```diff
@@ -183,6 +183,91 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 	}, nil
 }
 
+// ToClientWithHost creates an HTTP client.
+func (hcs *HTTPClientSettings) ToClientWithHost(host component.Host, settings component.TelemetrySettings) (*http.Client, error) {
```

Comment:
We need to deprecate `ToClient` in favor of this function (see paragraph about "deprecation notice" [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#breaking-changes))

Reply:
Done